pkg/storage/utils/indexer/indexer.go

```diff
@@ -166,9 +166,14 @@ func (i *StorageIndexer) FindBy(t interface{}, queryFields ...Field) ([]string,
 
 	resultPaths := make(map[string]struct{})
 	if fields, ok := i.indices[typeName]; ok {
-		for _, field := range queryFields {
-			for _, idx := range fields.IndicesByField[strcase.ToCamel(field.Name)] {
-				res, err := idx.Lookup(field.Value)
+		for fieldName, queryFields := range groupFieldsByName(queryFields) {
+			idxes := fields.IndicesByField[strcase.ToCamel(fieldName)]
+			values := make([]string, 0, len(queryFields))
+			for _, f := range queryFields {
+				values = append(values, f.Value)
+			}
+			for _, idx := range idxes {
+				res, err := idx.LookupCtx(context.Background(), values...)
```

Comment:
here we no longer make one Lookup call per queryField but instead aggregate by field name and then use the new LookupCtx to fetch all results at once.

I introduced the context because we want to be able to pass the trace context into the index as well ...